stdlib/typing.pyi

```diff
@@ -419,32 +419,16 @@ class Generator(Iterator[_YieldT_co], Generic[_YieldT_co, _SendT_contra, _Return
     @abstractmethod
     def throw(self, __typ: BaseException, __val: None = None, __tb: TracebackType | None = None) -> _YieldT_co: ...
     def close(self) -> None: ...
-    def __iter__(self) -> Generator[_YieldT_co, _SendT_contra, _ReturnT_co]: ...
```

Comment:
I'm unsure as to whether or not this should be -> Self the runtime has it as that in Iterator

Reply:
Upon further consideration, not implementing it as returning Self is probably a bug so I'll change it in (Async)Iterator even

Reply:
`typing.Generator.__iter__` shouldn't be annotated as returning `Self` for the same reason that `typing.Iterator.__iter__` shouldn't be annotated as returning `Self`. 99% of all iterators have `__iter__` methods that return `Self`, and 99% of generators are the same. But if we hinted them as such in the stub, we'd be changing the very _definition_ of Iterator/Generator such that the remaining 1% of iterators and generators would no longer be considered iterators or generators. And that wouldn't be accurate: it's not part of the contract that iterators have to return `self` from their `__iter__` methods at runtime in order for them to be considered iterators, they just have to return an iterable object.

With `types.GeneratorType`, it's a different issue, because that class doesn't define an interface in the same way. We can just reflect what happens at runtime there.

Reply:
I suspect removing this method is causing the mypy-primer fallout. Let's add it back.